src/libsyntax/ext/expand.rs

```diff
@@ -70,10 +70,13 @@ pub fn expand_expr(e: P<ast::Expr>, fld: &mut MacroExpander) -> P<ast::Expr> {
 
         ast::ExprKind::InPlace(placer, value_expr) => {
             // Ensure feature-gate is enabled
-            feature_gate::check_for_placement_in(
-                fld.cx.ecfg.features,
-                &fld.cx.parse_sess.span_diagnostic,
-                expr_span);
+            if let Some(&Features { pushpop_unsafe: false, .. }) =
+                fld.cx.ecfg.features {
```

Comment:
Could this be `if fld.cx.ecfg.features.unwrap().placement_in_syntax {`?
I think it would be better to panic if `features` were erroneously `None` for some reason than to silently bypass gated feature checks.